### PR TITLE
Fix GLTF animations

### PR DESCRIPTION
**Pull request by VitoTringolo**
Fix issues described in https://github.com/raysan5/raylib/issues/4055

### CODE REVIEW

File: src/raymath.h [hunk 1]
```diff
@@ -2525,4 +2525,54 @@ RMAPI int QuaternionEquals(Quaternion p, Quaternion q)
     return result;
 }
 
+// Decompose a transformation matrix into its rotational, translational and scaling components
+RMAPI void DecomposeMatrix(const Matrix *mat, Vector3 *translation, Quaternion *rotation, Vector3 *scale)
```
Comment (raysan5) [inline src/raymath.h:2529]: Note that this function is not following `raymath` conventions (naming, structure), it's going to be reviewed.

File: src/rmodels.c [hunk 2]
```diff
@@ -4008,6 +4008,37 @@
...
+
+static void NodeTransformWorld(cgltf_node *node, Matrix *result)
+{
+    cgltf_float worldTransform[16];
+    cgltf_node_transform_world(node, worldTransform);
+
+    CgltfMatrixToRLMatrix(worldTransform, result);
+}
+
+static void NodeTransformLocal(cgltf_node *node, Matrix *result)
```
[the remark this answers is not in the document]
Comment (raysan5) [reply src/rmodels.c:4032]: @VitoTringolo Please, could you review this issue?

File: src/rmodels.c [hunk 3]
```diff
@@ -5113,16 +5144,8 @@ static Model LoadGLTF(const char *fileName)
...
-
-            Matrix worldMatrix = {
-                worldTransform[0], worldTransform[4], worldTransform[8], worldTransform[12],
-                worldTransform[1], worldTransform[5], worldTransform[9], worldTransform[13],
-                worldTransform[2], worldTransform[6], worldTransform[10], worldTransform[14],
-                worldTransform[3], worldTransform[7], worldTransform[11], worldTransform[15]
-            };
-
+            Matrix worldMatrix;
+            NodeTransformWorld(node, &worldMatrix);
```
Comment (raysan5) [inline src/rmodels.c:5148]: Personally I prefer a bit of duplicate code than adding new functions.

File: src/raymath.h (hunk 1 above)
Comment (VitoTringolo) [reply src/raymath.h:2529]: Sorry I also read the guideline but I missed this one :/

**Comment by VitoTringolo** [comment]
OK @raysan5 I removed the functions and update the MatrixDecompose signature :)

**Comment by raysan5** [comment]
@VitoTringolo thanks for the review! this is a great addition! thanks!

**Comment by MrScautHD** [comment]
Hi, @VitoTringolo do you maybe have discrod?

Im doing my own Model Animation system and i has the same issue, and im not really sure how you fixed it.

**Pull request merged**